R/middleware.R

```diff
@@ -341,9 +350,11 @@ HandlerManager <- R6Class("HandlerManager",
           headers$'Content-Type' <- response$content_type
 
           response <- filter(req, response)
-          return(list(status=response$status,
-            body=response$content,
-            headers=headers))
+          return(list(
+            status = response$status,
+            body = if (head_request) "" else response$content,
```

Comment:
You're returning a `Content-Length` of `0`, which is misleading. Since you've done the work of generating the content, use its size to compute the proper content length.